crates/wit-parser/src/lib.rs

```diff
@@ -403,6 +403,21 @@ impl WorldItem {
     }
 }
 
+#[cfg_attr(feature = "serde", derive(Serialize))]
+#[derive(Debug, Clone)]
+pub struct Nest {
+    #[cfg_attr(feature = "serde", serde(serialize_with = "serialize_id"))]
+    pub id: InterfaceId,
+    pub docs: Docs,
+    #[cfg_attr(
+        feature = "serde",
+        serde(skip_serializing_if = "Stability::is_unknown")
+    )]
+    pub stability: Stability,
+    pub package_name: PackageName,
+    pub iface_name: String,
```

Comment:
I think both of these fields can be elided in favor of looking them up through the `id` here

Reply:
So I realized that the `id` is actually not being updated correctly, I believe when [append](https://github.com/bytecodealliance/wasm-tools/blob/main/crates/wit-parser/src/resolve.rs#L461) is being called.  I've spent some time trying to figure out how to update this, but haven't quite worked it out yet.  I thought it might be worth considering whether it would be ok to get rid of the `id` though and use the fields instead.  I see the appeal of using `id`, but I do think that might complicate what it means to update interfaces when combining packages, unless it's clear to you how to simply perform the update.